osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/application/config/messaging/OutboundLogItemRequestsMessagingConfig.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec61850.application.config.messaging;
+
+import javax.jms.ConnectionFactory;
+import javax.net.ssl.SSLException;
+
+import org.opensmartgridplatform.adapter.protocol.iec61850.infra.messaging.Iec61850LogItemRequestMessageSender;
+import org.opensmartgridplatform.adapter.protocol.iec61850.services.DeviceMessageLoggingService;
+import org.opensmartgridplatform.shared.application.config.messaging.DefaultJmsConfiguration;
+import org.opensmartgridplatform.shared.application.config.messaging.JmsConfigurationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.jms.core.JmsTemplate;
+
+/**
+ * Configuration class for outbound log item requests.
+ */
+@Configuration
+public class OutboundLogItemRequestsMessagingConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OutboundLogItemRequestsMessagingConfig.class);
+
+    private JmsConfigurationFactory jmsConfigurationFactory;
+
+    public OutboundLogItemRequestsMessagingConfig(final Environment environment,
+            final DefaultJmsConfiguration defaultJmsConfiguration) throws SSLException {
+        this.jmsConfigurationFactory = new JmsConfigurationFactory(environment, defaultJmsConfiguration,
+                "jms.iec61850.log.item.requests");
+    }
+
+    @Bean(destroyMethod = "stop", name = "protocolIec61850OutboundLogItemRequestsConnectionFactory")
+    public ConnectionFactory connectionFactory() {
+        LOGGER.info("Initializing protocolIec61850OutboundLogItemRequestsConnectionFactory bean.");
+        return this.jmsConfigurationFactory.getPooledConnectionFactory();
+    }
+
+    @Bean(name = "protocolIec61850OutboundLogItemRequestsJmsTemplate")
+    public JmsTemplate jmsTemplate() {
+        LOGGER.info("Initializing protocolIec61850OutboundLogItemRequestsJmsTemplate bean.");
+        return this.jmsConfigurationFactory.initJmsTemplate();
+    }
+
+    // TODO - Fix "Classes with only "static" methods should not be instantiated
+    // (squid:S2440)" sonar warning
+    @Bean
+    public DeviceMessageLoggingService deviceMessageLoggingService(
+            @Qualifier("protocolIec61850OutboundLogItemRequestsMessageSender") final Iec61850LogItemRequestMessageSender messageSender) {
+        return new DeviceMessageLoggingService(messageSender);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this instantiation of "DeviceMessageLoggingService". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS2440)